### PR TITLE
Require openGL context 2.1

### PR DESCRIPTION
**Pull request by atupone**
Requesting openGL2.1

Removing all the check for things that are mandatory for that openGL version

### CODE REVIEW

File: src/bzflag/BackgroundRenderer.cxx [hunk 1]
```diff
@@ -853,8 +847,8 @@ void BackgroundRenderer::drawSkybox()
     if (!BZDBCache::drawGround)
     {
         tm.bind(skyboxTexID[5]); // bottom
-        glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, skyboxWrapMode);
-        glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, skyboxWrapMode);
+        glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
+        glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
         glBegin(GL_TRIANGLE_STRIP);
```
Comment (jwmelto) [inline src/bzflag/BackgroundRenderer.cxx]: This is begging for a function (or at least a lambda) to reduce the amount of copy/paste

File: src/bzflag/bzflag.cxx [hunk 2]
```diff
@@ -1288,7 +1288,7 @@ int         main(int argc, char** argv)
             case GL_OUT_OF_MEMORY:
                 std::cerr << "ERROR: GL_OUT_OF_MEMORY" << std::endl;
                 break;
-#ifdef GL_VERSION_1_2
+#ifdef GL_TABLE_TOO_LARGE
```
Comment (jwmelto) [inline src/bzflag/bzflag.cxx]: is this optional in GL 2.1? If not, we can probably do away with the `#ifdef`

Comment (atupone) [reply src/bzflag/bzflag.cxx]: That is eventually for GLES. GL_TABLE_TOO_LARGE is not applicable to it

**Comment by atupone** [comment]
Before merging I would like to see if some of the code that I removed still apply to the eventual GLES porting.

File: src/bzflag/bzflag.cxx (hunk 2 above)
Comment (macsforme) [reply src/bzflag/bzflag.cxx]: When that time comes, we should find a better test than that (and there will need to be many others). Compiler errors will show any unsupported code that needs to be updated. In my GLES port, I just defined `HAVE_GLES`, which also made it easy to find any GLES-specific code. I suggest just remove that `#ifdef` for now.

Comment (atupone) [reply src/bzflag/bzflag.cxx]: lot of code out of here is using that ifdef.
But I don't care removing that for now

File: src/bzflag/BackgroundRenderer.cxx (hunk 1 above)
Comment (atupone) [reply src/bzflag/BackgroundRenderer.cxx]: It seems the pattern for introducing a 3d texture there

**Pull request merged**